tests/playwright/src/model/pages/resources-page.ts

```diff
@@ -34,5 +40,26 @@ export class ResourcesPage extends SettingsPage {
     this.podmanResources = this.featuredProviderResources.getByRole('region', { name: 'podman', exact: true });
     this.composeResources = this.featuredProviderResources.getByRole('region', { name: 'Compose', exact: true });
     this.kindResources = this.featuredProviderResources.getByRole('region', { name: 'kind', exact: true });
+    this.kindProviderSetup = this.kindResources.getByRole('region', { name: 'Provider Setup' });
+    this.kindProviderConnections = this.kindResources.getByRole('region', { name: 'Provider Connections' });
+    this.kindClusterStatus = this.kindProviderConnections.getByLabel('Connection Status Label');
+    this.kindClusterCreationButton = this.kindResources.getByRole('button', { name: 'Create new Kind Cluster' });
+  }
+
+  public async openClusterCreationPage(): Promise<CreateKindClusterPage> {
+    await playExpect(this.kindResources).toBeVisible();
+    await this.kindClusterCreationButton.click();
+    return new CreateKindClusterPage(this.page);
+  }
+
+  public async kindClusterExists(label = 'kind-cluster'): Promise<boolean> {
+    return (await this.kindProviderConnections.getByRole('region', { name: label }).count()) > 0;
+  }
+
+  public async operateWithKindCluster(operation: string): Promise<void> {
+    await playExpect(
+      this.kindProviderConnections.getByRole('button', { name: operation, exact: true }),
+    ).not.toBeDisabled({ timeout: 7000 });
+    await this.kindProviderConnections.getByRole('button', { name: operation, exact: true }).click();
```

Comment:
If you are going to implement such a general method the param for it should be a value from an enum, not just a plan string, otherwise users won't know what possible values there are.